src/time_utils.c

```diff
@@ -125,3 +126,33 @@ ts_time_value_from_arg(Datum arg, Oid argtype, Oid timetype)
 
 	return ts_time_value_to_internal(arg, argtype);
 }
+
+int64
+ts_time_get_nobegin(Oid timetype)
+{
+	int64 nobegin;
+
+	switch (timetype)
+	{
+		case DATEOID:
+			nobegin = DATEVAL_NOBEGIN;
+			break;
```

Comment:
Nit: This format is significantly easier to follow:
```suggestion
		case DATEOID:
			return DATEVAL_NOBEGIN;
```

You can also use an array inside the function, which might be easier to maintain.

Reply:
Not sure I understand the array thing. Can you clarify? Please note that the enum values here are not, e.g., 0, 1, 2... For example, `TIMESTAMPTZOID` is 1114.

Reply:
Yeah, right. I was thinking of smaller values. Sorry about that.